### PR TITLE
[python3] x64-linux-dynamic fixes

### PR DESCRIPTION
**Pull request by Osyotr**
This PR fixes build issues with `x64-linux-dynamic` triplet.
1) Patch `setup.py` to force rpath flags for extensions.
2) Implement some post-build checks to ensure extensions built successfully.
3) Set `VCPKG_POLICY_MISMATCHED_NUMBER_OF_BINARIES` because there's an additional binary in release builds (see comment in portfile).
4) Add missing `libuuid` dependency

For https://github.com/microsoft/vcpkg/issues/25668

### CODE REVIEW

**Review by Adela0814 — DISMISSED** [review]

File: ports/python3/portfile.cmake [hunk 1]
```diff
@@ -244,6 +250,19 @@ else()
...
 
     vcpkg_fixup_pkgconfig()
+
+    file(GLOB python_libs_dynload_release LIST_DIRECTORIES false "${CURRENT_PACKAGES_DIR}/lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/lib-dynload/*_failed.so*")
+    file(GLOB python_libs_dynload_debug LIST_DIRECTORIES false "${CURRENT_PACKAGES_DIR}/debug/lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/lib-dynload/*_failed.so*")
+    if(python_libs_dynload_release OR python_libs_dynload_debug)
+        message(FATAL_ERROR "There should be no modules with \"_failed\" suffix")
+    endif()
+    if(NOT VCPKG_BUILD_TYPE)
+        list(LENGTH python_libs_dynload_release python_libs_dynload_release_length)
```
Comment (Osyotr) [inline ports/python3/portfile.cmake:260]:
```suggestion
        file(GLOB python_libs_dynload_release LIST_DIRECTORIES false "${CURRENT_PACKAGES_DIR}/lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/lib-dynload/*.so*")
        file(GLOB python_libs_dynload_debug LIST_DIRECTORIES false "${CURRENT_PACKAGES_DIR}/debug/lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/lib-dynload/*.so*")
        list(LENGTH python_libs_dynload_release python_libs_dynload_release_length)
```

Comment (Adela0814) [reply ports/python3/portfile.cmake:260]: Why don't you merge this into the above?
```
    file(GLOB python_libs_dynload_release LIST_DIRECTORIES false "${CURRENT_PACKAGES_DIR}/lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/lib-dynload/*_failed.so*")
    file(GLOB python_libs_dynload_debug LIST_DIRECTORIES false "${CURRENT_PACKAGES_DIR}/debug/lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/lib-dynload/*_failed.so*")
```

**Comment by Osyotr** [comment]
@Hoikas is it worth adding `VCPKG_POLICY_MISMATCHED_NUMBER_OF_BINARIES` just to keep python stable abi library (`libpython3.so`)?

**Comment by Hoikas** [comment]
I'm not sure how widely the limited API library is used, TBH. If you have a use for it or it's trivial to make available, I think it would be a good thing to have as an option.

**Comment by Osyotr** [comment]
It's already available but with the cost of `VCPKG_POLICY_MISMATCHED_NUMBER_OF_BINARIES` :(
I don't use it myself and haven't found any projects that actually use it. Both boost-python and SWIG haven't adopted it yet.
https://github.com/boostorg/python/issues/221
https://github.com/swig/swig/pull/2190

**Comment by Osyotr** [comment]
@Adela0814 maybe `vcpkg-team-review` this?

**Review by Adela0814 — DISMISSED** [review]

File: ports/python3/portfile.cmake [hunk 2]
```diff
@@ -204,6 +205,11 @@ if(VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_UWP)
         endif()
     endif()
 else()
+    # Python Stable ABI is incompatible with --with-pydebug option
```
Comment (BillyONeal) [inline ports/python3/portfile.cmake:208]: vcpkg does not provide a stable ABI for installed trees, so I'm not sure why this is a relevant distinction? Note that vcpkg's python almost never matches the system python.

**Review by BillyONeal — CHANGES_REQUESTED** [review]
>Set VCPKG_POLICY_MISMATCHED_NUMBER_OF_BINARIES because there's an additional binary in release builds (see comment in portfile).

Can you explain what the additional binary is?

**Comment by Osyotr** [comment]
@BillyONeal the additional binary is `libpython3.so` aka "Python Stable ABI". It's not created in debug builds.

File: ports/python3/portfile.cmake (hunk 2 above)
Comment (Osyotr) [reply ports/python3/portfile.cmake:208]: This is what their buildsystem does.
Upon further investigation, I found out that debug builds are now abi compatible with release builds https://github.com/python/cpython/pull/12615
Maybe ask maintainers whether this is something they forgot to address in their buildsystem?

Comment (BillyONeal) [reply ports/python3/portfile.cmake:208]: Ah, so this isn't

**Comment by BillyONeal** [comment]
> @BillyONeal the additional binary is `libpython3.so` aka "Python Stable ABI". It's not created in debug builds. https://peps.python.org/pep-0384/#linkage

Ah, I see. "Python Stable ABI" isn't a statement about the shipping Python bits, it's the name of a specific binary.

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

File: ports/python3/0011-dont-skip-rpath.patch [hunk 3]
```diff
@@ -0,0 +1,158 @@
+diff --git a/setup.py b/setup.py
+index 61b3266..4c2cfb4 100644
+--- a/setup.py
++++ b/setup.py
+@@ -1153,6 +1153,7 @@ def detect_readline_curses(self):
+             self.add(Extension('readline', ['readline.c'],
+                                library_dirs=['/usr/lib/termcap'],
+                                extra_link_args=readline_extra_link_args,
++                               runtime_library_dirs=self.lib_dirs,
```
Comment (dg0yt) [inline ports/python3/0011-dont-skip-rpath.patch:9]: Apart from using absolute paths, this patch broke the static build. The build ends successful, but it lacks important components such as `binascii`. The error logs lists unknown `-R...` parameters.

Comment (Osyotr) [reply ports/python3/0011-dont-skip-rpath.patch:9]: Use of absolute paths is ok at build time. They will be fixed when installing.
I'll try to get rid of this patch later today and disable python's check for whether module loads or not.

**Pull request merged**